src/AppBundle/Controller/BlogController.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace AppBundle\Controller;
+
+use Symfony\Bundle\TwigBundle\TwigEngine;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use eZ\Publish\API\Repository\SearchService;
+use AppBundle\QueryType\ChildrenQueryType;
+
+class BlogController
+{
+    /** @var \Symfony\Bundle\TwigBundle\TwigEngine */
+    private $templating;
+
+    /** @var \eZ\Publish\API\Repository\SearchService */
+    private $searchService;
+
+    /** @var \AppBundle\QueryType\ChildrenQueryType */
+    private $childrenQueryType;
+
+    /** @var int */
+    private $blogLocationId;
+
+    /** @var int */
+    private $blogPostsLimit;
+
+    /**
+     * @param \Symfony\Bundle\TwigBundle\TwigEngine $templating
+     * @param \eZ\Publish\API\Repository\SearchService $searchService
+     * @param \AppBundle\QueryType\ChildrenQueryType $childrenQueryType
+     * @param int $blogLocationId
+     * @param int $blogPostsLimit
+     */
+    public function __construct(
+        TwigEngine $templating,
+        SearchService $searchService,
+        ChildrenQueryType $childrenQueryType,
+        $blogLocationId,
+        $blogPostsLimit
+    ) {
+        $this->templating = $templating;
+        $this->searchService = $searchService;
+        $this->childrenQueryType = $childrenQueryType;
+        $this->blogLocationId = $blogLocationId;
+        $this->blogPostsLimit = $blogPostsLimit;
+    }
+
+    /**
+     * Renders `blog_post` list for the given $page.
+     *
+     * @param int $page
+     *
+     * @return \Symfony\Component\HttpFoundation\JsonResponse
+     */
+    public function showBlogPostsAction($page)
+    {
+        $offset = $page * $this->blogPostsLimit - $this->blogPostsLimit;
+
+        $query = $this->childrenQueryType->getQuery([
+            'parent_location_id' => $this->blogLocationId,
+            'limit' => $this->blogPostsLimit,
+            'offset' => $offset,
+        ]);
+        $searchResults = $this->searchService->findLocations($query);
+
+        $renderedContent = $this->templating->render('parts/content_list.html.twig', [
+            'items' => $searchResults,
```

Comment:
No we can't, it's related to how `QueryTypes` works when injected for the specific view.